### DISCUSSION TITLE
Include `test/VFSTests.js` in buildless branch?

### DISCUSSION THREAD

**Discussion by LoneRifle** [rhashimoto/wa-sqlite#60] [Ideas]
Hello everyone,

I'm in the middle of porting [sqlite-s3vfs](https://github.com/uktrade/sqlite-s3vfs) - a Python VFS written by @uktrade - to TypeScript for wa-sqlite. This is a VFS that uses S3 as the persistent store, and I feel that a TypeScript port would unlock very interesting usecases. In particular, serverless functions that can only be written in JavaScript/TypeScript and not have native modules (eg. Netlify Functions) stand to benefit, since they can now use S3-compatible stores as a cloud-hosted relational database.

I would love to make use of `test/VFSTests.js` to verify the behaviour of what I've [made](https://github.com/LoneRifle/wa-sqlite-s3vfs), as would many other authors of third-party VFSes.

@rhashimoto - would you be happy to add the test suite as part of your build artefacts or somehow make it available? Would love to hear what you think!

**Comment by rhashimoto** [comment]
I'm not quite sure what you're asking for. Since VFSTests.js already exists on the buildless branch I think it's simply adding the test file(s) to the `files` section of package.json, is that right? If yes, I'd be happy to do that. If no, perhaps you could help me understand the issue better.

Off-topic, I noticed that you chose the MIT license for your VFS project, but I would caution you that I don't think that a wa-sqlite VFS would qualify for MIT licensing since the default wa-sqlite license is GPL. This is a complicated question that ultimately can only be decided by a court ruling, but [my reading of the FSF's opinions](https://github.com/kikko-land/kikko/discussions/35#discussioncomment-3174623) is escaping the GPL here is borderline at best.

**Reply by LoneRifle** [reply]
Apologies, should have been more precise in my wording; yes, if you could add the file in question to the `files` manifest, I would be grateful. I'll change my license to be in-line with yours!

**Reply by rhashimoto** [reply]
Done. Be aware that you might need to increase the jasmine timeout (e.g. like [this](https://github.com/rhashimoto/wa-sqlite/blob/b73b14678d48f6e4d3afdd7838ecb85cdfe5e855/test/OriginPrivateFileSystemVFS.test.js#L8)) for a slower VFS. The [contention test](https://github.com/rhashimoto/wa-sqlite/blob/b73b14678d48f6e4d3afdd7838ecb85cdfe5e855/test/VFSTests.js#L325-L402), which attempts to exercise concurrent access with 8 connections, might be particularly slow.

**Comment by twoxfh** [comment]
Curious, my understanding is the application which uses wa-sqlite has to be GPL also. Does that mean the UK government has opensourced the application that uses wa-sqlite and the vfs addition? If so, any link?

**Reply by LoneRifle** [reply]
The UK Government has not used wa-sqlite at all =) what they have published is a VFS written in Python for APSW, a sqlite integration for Python
